### PR TITLE
Move RequestIdentifierFeature to HttpContext

### LINKED ISSUE #412: Expose TraceIdentifier as a property on HttpContext
It's pretty convenient to make available.

### PR DESCRIPTION
**Pull request by benaadams**
Rebased #435
Allow lazier instantiation
Expose TraceIdentifier on Httpcontext
Also resolves #412

Add tests

### CODE REVIEW

File: src/Microsoft.AspNet.Http/DefaultHttpContext.cs [hunk 1]
```diff
@@ -118,6 +118,19 @@ private ISessionFeature SessionFeature
...
         }
 
+        private IHttpRequestIdentifierFeature RequestIdentifierFeature
+        {
+            get {
+                return FeatureHelpers.GetOrCreate<IHttpRequestIdentifierFeature>(
+                  _features,
+                  () => new HttpRequestIdentifierFeature());
+            }
+            set
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http/DefaultHttpContext.cs]: This setter is never used, correct? Remove it.

**Comment by Tratcher** [comment]
:shipit: after my minor comment.

File: src/Microsoft.AspNet.Http/DefaultHttpContext.cs (hunk 1 above)
Comment (benaadams) [inline src/Microsoft.AspNet.Http/DefaultHttpContext.cs]: Removed, squished

**Comment by benaadams** [comment]
Applied feedback, squished

**Pull request merged**